### PR TITLE
fix: Allow more generic Code Type

### PR DESCRIPTION
**Pull request by romain-tadiello**
This PR implements a more free type Code with its dedicated unmarshaling function.
From https://outreach-io.atlassian.net/browse/ORES-2170

### CODE REVIEW

File: errors.go [hunk 1]
```diff
@@ -21,6 +21,31 @@ type ErrorsPayload struct {
 	Errors []*ErrorObject `json:"errors"`
 }
 
+// Code represents the application-specific error code, expressed as a string value.
+type Code string
+
+// UnmarshalJSON converts the JSON (number or strign) code into string Code
```
Comment (antonaudzei-outreach) [inline errors.go]:
```suggestion
// UnmarshalJSON converts the JSON number or string value into Code string
```

File: errors.go [hunk 2]
```diff
@@ -21,6 +21,31 @@ type ErrorsPayload struct {
...
+	case float64:
+		*c = Code(fmt.Sprintf("%.0f", v))
+	default:
+		return fmt.Errorf("json: cannot unmarshal into Go value of type %T ", v)
+	}
+	return nil
+}
+
+// ToString returns code as a string
+func (c *Code) ToString() string {
```
Comment (antonaudzei-outreach) [inline errors.go]: Not sure if this function is useful. But if it is I think it needs to become
```
func (c Code) ToString() string {
```
pointer is unnecessary

**Review by antonaudzei-outreach — APPROVED** [review]
Approved with suggestions

File: errors.go (hunk 2 above)
Comment (romain-tadiello) [reply errors.go]: That is a bit overkill I agree, I will keep it simple for now and just remove this method

**Pull request merged**